### ISSUE TITLE
rubocop autocorrect throws exception for Style/GuardClause.

### ISSUE THREAD

**Issue by MrRaffnix** [rubocop/rubocop#11291]
## Expected behavior

No exception thrown for autocorrection of cop `Style/GuardClause`  when `rubocop -A` is executed. All other autocorrection seems to work.

## Actual behavior

When running `rubocop -A` the following exception occurs:

```bash
An error occurred while Style/GuardClause cop was inspecting /Users/anonymus/src/some_app/lib/mergeable_configatron.rb:93:8.
undefined method `heredoc?' for s(:const, nil, :UnknownProtectedKeyException):RuboCop::AST::ConstNode
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cop/style/guard_clause.rb:190:in `autocorrect'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cop/style/guard_clause.rb:179:in `block in register_offense'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cop/base.rb:346:in `correct'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cop/base.rb:127:in `add_offense'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cop/style/guard_clause.rb:176:in `register_offense'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cop/style/guard_clause.rb:128:in `on_if'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:100:in `public_send'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:100:in `block (2 levels) in trigger_responding_cops'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:160:in `with_cop_error_handling'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:99:in `block in trigger_responding_cops'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:98:in `each'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:98:in `trigger_responding_cops'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:69:in `on_if'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-ast-1.24.0/lib/rubocop/ast/traversal.rb:157:in `on_if'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:71:in `on_if'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-ast-1.24.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-ast-1.24.0/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-ast-1.24.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-ast-1.24.0/lib/rubocop/ast/traversal.rb:158:in `on_block'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:71:in `on_block'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-ast-1.24.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-ast-1.24.0/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-ast-1.24.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-ast-1.24.0/lib/rubocop/ast/traversal.rb:154:in `on_def'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:71:in `on_def'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-ast-1.24.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-ast-1.24.0/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-ast-1.24.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-ast-1.24.0/lib/rubocop/ast/traversal.rb:154:in `on_class'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:71:in `on_class'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-ast-1.24.0/lib/rubocop/ast/traversal.rb:20:in `walk'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:86:in `investigate'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cop/team.rb:154:in `investigate_partial'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cop/team.rb:82:in `investigate'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:315:in `inspect_file'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:259:in `block in do_inspection_loop'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:293:in `block in iterate_until_no_changes'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:286:in `loop'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:286:in `iterate_until_no_changes'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:255:in `do_inspection_loop'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:138:in `block in file_offenses'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:163:in `file_offense_cache'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:137:in `file_offenses'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:128:in `process_file'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:109:in `block in each_inspected_file'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:108:in `each'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:108:in `reduce'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:108:in `each_inspected_file'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:94:in `inspect_files'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:47:in `run'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cli/command.rb:11:in `run'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cli/environment.rb:18:in `run'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cli.rb:72:in `run_command'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cli.rb:79:in `execute_runners'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/lib/rubocop/cli.rb:48:in `run'
/Users/anonymus/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.40.0/exe/rubocop:19:in `block in <top (required)>'
```

## Steps to reproduce the problem

The rubocop autocorrect is failing for a few files, here to show:
```ruby
class CountryCodeValidator < ActiveModel::EachValidator
  def validate_each(record, attribute, value)
    unless ISO3166::Data.codes.include?(value)
      record.errors.add(attribute,
                        I18n.t("activerecord.errors.models.account/practice.attributes.#{attribute}.inclusion"))
    end
  end
end
```
```ruby
require 'uploaded_string_io'

class Comm::Messaging::Postmaster
  class UnmatchedMessageError < ActiveRecord::RecordNotFound
  end

  class CannotDeliverMessageError < ArgumentError
  end

  class << self
    # delivers the message to the user
    #
    # this is the only public method for delivering internal messages.
    # it distincts between sms and normal messages. sms messages are
    # sent via sms, normal messages are put into the message box of the
    # recipient (therefore the recipients needs to be a user)
    #
    # == Parameters
    #   message<Message, Hash>: the message to deliver
    def deliver!(message)
      m = delivery_method(message)

      if m.present?
        send(m, message)
      else
        raise CannotDeliverMessageError, "Can't deliver message! Message was #{message.inspect}"
      end
    end

    # returns whether or not the message can be delivered
    def can_deliver_message?(message)
      delivery_method(message).present?
    end

    # returns the method to call which can deliver the given message
    # returns nil when the message cannot be delivered
    def delivery_method(message)
      recipient = message.recipient

      if message.sms?
        :deliver_as_sms
      elsif message.email?
        :deliver_as_email
      elsif recipient.respond_to?(:message_box) && recipient.message_box.present?
        :deliver_to_messagebox
      elsif recipient.is_a?(Patient::Patient) && recipient.is_user?
        raise 'not implemented' # replaced by MessageSender
      end
    end
  end
end

```

## RuboCop version
```
1.40.0 (using Parser 3.1.3.0, rubocop-ast 1.24.0, running on ruby 2.7.6) [arm64-darwin21]
  - rubocop-minitest 0.24.0
  - rubocop-performance 1.15.1
  - rubocop-rails 2.17.3
  - rubocop-rspec 2.15.0
 ```

**Comment by koic** [comment]
This issue is a dup and it has been solved by #11250.

**Comment by MrRaffnix** [comment]
Thanks and sorry for not seeing it 😢
